src/Backtracking.jl

```diff
@@ -0,0 +1,58 @@
+#=
+Backtracking line search implementation
+=#
+
+#=
+Input parameters:
+    - x: current iterate
+    - g: gradient 
+    - α: initial step size
+    - p: descent direction
+    - f: function evaluation
+=#
+
+mutable struct Backtracking{X, F, A}
+    x::X
+    g::X
+    p::X
+    f::F
+    α::A
+end
+
+# Determine step size by backtracking line search
+function Backtracking(backtrack::Backtracking)
+    # Setup (Function information)
+    x = backtrack.x
+    g = backtrack.g
+    α = backtrack.α
+    p = backtrack.p
+    f = backtrack.f
+
+    # Setup (Parameters)
+    c1 = 1e-4
+    β = 0.5
+    iterations = 1_000
+
+    ϕ_0 = f(x)
+    ϕ_α = f(x + α*p)
+    iteration = 0
+
+    # Backtracking line search
+    while ϕ_α > ϕ_0 + c1 * α * g * p
```

Comment:
Move this oput to a separate function that takes as argument the function and a `Backtracking` object